Jenkinsfile

```diff
@@ -1,19 +1,101 @@
-<?xml version="1.0" encoding="UTF-8" ?>
-<!DOCTYPE ci [
-  <!ELEMENT ci (task*,matrix*)>
-  <!ELEMENT task (#PCDATA)>
-  <!ATTLIST task id ID #REQUIRED>
-  <!ELEMENT matrix (run*)>
-  <!ATTLIST matrix id ID #REQUIRED>
-  <!ELEMENT run (v*)>
-  <!ATTLIST run matrix IDREF #IMPLIED>
-  <!ATTLIST run task IDREF #IMPLIED>
-  <!ELEMENT v (#PCDATA)>
-  <!ATTLIST v n CDATA #REQUIRED>
-]>
-<ci>
-
-  <task id="main"><![CDATA[
+// If not a PR, this is a long-lived branch, which should have a nightly build
+def triggers = []
+if (!env.CHANGE_ID) {
+  // This is the 0.6.x series: run weekly on Sunday
+  triggers << cron('H H(0-2) * * 6')
+}
+
+// Setup properties of this job definition
+properties([
+  parameters([
+    string(name: 'matrix', defaultValue: 'auto', description: 'The matrix to build (auto, quick, full)')
+  ]),
+  pipelineTriggers(triggers)
+])
+
+// Check whether the job was started by a timer
+// See https://hopstorawpointers.blogspot.ch/2016/10/performing-nightly-build-steps-with.html
+@NonCPS
+def isJobStartedByTimer() {
+  def startedByTimer = false
+  def buildCauses = currentBuild.rawBuild.getCauses()
+  for (buildCause in buildCauses) {
+    if (buildCause != null) {
+      def causeDescription = buildCause.getShortDescription()
+      echo "shortDescription: ${causeDescription}"
```

Comment:
Yes, it's useful as a debugging message if anything doesn't go as expected (e.g., an update of Jenkins that would change that string ..., since it's not specified).